packages/next-on-pages/templates/_worker.js/utils/cache-interface.ts

```diff
@@ -0,0 +1,248 @@
+export const SUSPENSE_CACHE_URL = 'INTERNAL_SUSPENSE_CACHE_HOSTNAME';
+
+/**
+ * Gets the cache interface to use for the suspense cache.
+ *
+ * Prioritises KV > D1 > R2 > Cache API.
+ *
+ * @returns Interface for the suspense cache.
+ */
+export async function getSuspenseCacheInterface(): Promise<CacheInterface> {
+	if (process.env.KV_SUSPENSE_CACHE) {
+		return new KVCacheInterface(process.env.KV_SUSPENSE_CACHE);
+	}
+
+	if (process.env.D1_SUSPENSE_CACHE) {
+		return new D1CacheInterface(process.env.D1_SUSPENSE_CACHE);
+	}
+
+	if (process.env.R2_SUSPENSE_CACHE) {
+		return new R2CacheInterface(process.env.R2_SUSPENSE_CACHE);
+	}
+
+	const cacheApi = await caches.open('suspense-cache');
```

Comment:
I wonder if it might be better to use `caches.default` here, instead of opening a new one. Thoughts?

Reply:
I think it's totally fine to open a specific cache, would using the default one provide any benefit?

Reply:
> I think it's totally fine to open a specific cache, would using the default one provide any benefit?

I'm not really sure. I want to say it wouldn't